### ISSUE TITLE
Massive disk usage during cache rebuilding

### ISSUE THREAD

**Issue by chneukirchen** [borgbackup/borg#2515]
After .cache/borg got corrupted, I removed it, and touched `.cache/borg/.../chunks.archive.d`.
The next backup took a very long time, and downloaded roughly 20G of data into `/var/tmp`.
(I set TMPDIR=/var/tmp.)

Command line was `borg create -s -C zlib -e ... user@remote:foo.borg::foo-full-... /`

What remained are ~700MB `chunks` and ~200MB `files`, which is reasonable.

The repo is quite big, but the temporary data usage should not be 20x of the usual need.

Borg 1.0.10 running on both ends, on Linux x86_64 (source: Void, target: Arch).

**Comment by chneukirchen** [comment]
`borg info`:

```
Number of files: 2581935

                       Original size      Compressed size    Deduplicated size
This archive:              162.25 GB            112.37 GB            182.07 MB
All archives:               46.18 TB             29.38 TB            351.32 GB

                       Unique chunks         Total chunks
Chunk index:                 9445877            908066328
```

**Comment by enkore** [comment]
This is mainly due to the metadata caching mechanism being underdeveloped. When Borg syncs the cache, it caches retrieved metadata (so that archives that re-use the deduplicated metadata don't need to download it again), but that cache has no bound and no eviction policy.

**Comment by chneukirchen** [comment]
It also could be interesting to experiment whether keeping a cache of all chunks is useful, I guess a cache of the chunks seen last month/year already covers 99% of all hits...

The current problem is that the chunk cache grows linear with the repo, which makes long term usage difficult.

**Comment by ThomasWaldmann** [comment]
see also #2518.

**Comment by enkore** [comment]
Fixed by #2572
